gapic/src/main/com/google/gapid/perfetto/canvas/RenderContext.java

```diff
@@ -284,6 +284,45 @@ public void drawTextLeftTruncate(
     }
   }
 
+  // draws text centered vertically, right truncated to fit into the given width.
+  public void drawTextRightTruncate(
+      Fonts.Style style, String text, double x, double y, double w, double h) {
+    drawTextRightTruncate(style, text, x, y, w, h, false);
+  }
+
+  // draws text centered horizontally and vertically, right truncated to fit into the given width.
+  public void drawTextCenteredRightTruncate(
+      Fonts.Style style, String text, double x, double y, double w, double h) {
+    drawTextRightTruncate(style, text, x, y, w, h, true);
+  }
+
+  private void drawTextRightTruncate(
+      Fonts.Style style, String text, double x, double y, double w, double h, boolean centered) {
+    String toDisplay = text;
+    for (int l = text.length(); ; ) {
+      Size size = fontContext.measure(style, toDisplay);
+      if (size.w < w) {
+        drawText(style, toDisplay, x + (centered ? (w - size.w) / 2 : 0), y + (h - size.h) / 2);
+        break;
+      }
+
+      l = Math.min(l - textSizeGreediness, (int)(w / (size.w / toDisplay.length())));
+      if (l <= 0) {
+        break;
+      }
+      toDisplay = text.substring(0, l) + "...";
+    }
+  }
+
+  public void drawTextTruncated(
+      Fonts.Style style, String text, double x, double y, double w, double h, Truncate truncate) {
+    if (truncate == Truncate.Left) {
```

Comment:
I still think it should be a boolean than a two possibility enum. 
Doing a search about `public static enum` in AGI we can see that enum are mostly defined when there's a lot of options or when the items are highly descriptive.
We do `if(truncate == Truncate.Left) else` rather than `if(truncate == Truncate.Left) else if (truncate == Truncate.Right)` here further makes me feel boolean is enough for this task. 
I see the benefit of easy readibility at the panel creation time, but that part of code is not modified that often and it's easy to go one step further to see the meaning of the boolean value, so I still think boolean is better in this case.